src/sinks/elasticsearch.rs

```diff
@@ -291,7 +293,10 @@ impl ElasticSearchCommon {
         };
 
         let base_url = config.host.clone();
-        let region = region_from_endpoint(&config.host)?;
+        let region = match &config.aws {
+            Some(region) => Region::try_from(region)?,
+            None => region_from_endpoint(&config.host)?,
```

Comment:
Are there cases where this would incorrectly assume you're talking to the AWS service based on the hostname? E.g. would it try to use AWS authentication when talking to `us-east-1.lukes-es-service.com`?

Reply:
No. The choice of using AWS authentication is guarded by the `auth.strategy` configuration setting. This is just figuring out what region to use if it gets there.